cloudify/cluster.py

```diff
@@ -13,158 +13,26 @@
-    if dirname and not os.path.exists(dirname):
-        os.makedirs(dirname)
-
-    with open(filename, 'w') as f:
-        json.dump(settings, f, indent=4, sort_keys=True)
-
-
-def get_cluster_nodes(filename=None):
-    settings = get_cluster_settings(filename=filename)
-    if not settings:
-        return None
-    return settings.get('nodes')
-
-
-def set_cluster_nodes(nodes, filename=None):
-    settings = get_cluster_settings(filename=filename) or {}
-    if not settings and not nodes:
-        return
-    settings['nodes'] = nodes
-    set_cluster_settings(settings, filename=filename)
-    active = get_cluster_active(filename=filename)
-    if active is None:
-        set_cluster_active(nodes[0], filename=filename)
-    return nodes
-
-
-def get_cluster_active(filename=None):
-    settings = get_cluster_settings(filename=filename)
-    if not settings:
-        return None
-
-    active = settings.get('active')
-    if active:
-        return active
-    # when we don't know which node is the active, try the first one on the
-    # list - if it's a replica, we'll failover normally
-    nodes = get_cluster_nodes(filename=filename)
-    if nodes:
-        set_cluster_active(nodes[0], filename=filename)
-    return nodes[0]
-
-
-def set_cluster_active(node, filename=None):
-    settings = get_cluster_settings(filename=filename) or {}
-    if not settings and not node:
-        return
-    settings['active'] = node
-    set_cluster_settings(settings, filename=filename)
-
-
-def get_cluster_amqp_settings(filename=None):
-    settings = get_cluster_settings(filename=filename)
-
-    if settings:
-        return settings['nodes']['brokers']
-    else:
-        return []
-
-
-def delete_cluster_settings(filename=None):
-    """Remove all cluster settings.
-
-    Delete the settings file, and also all stored certificates - find the
-    certs dir first.
-    """
-    filename = _get_cluster_settings_file(filename)
-    if not filename:
-        return
-    try:
-        os.remove(filename)
-    except (OSError, IOError):
-        # the file doesn't exist?
-        pass
-
-
-def is_cluster_configured(filename=None):
-    path = _get_cluster_settings_file(filename=filename)
-    if not path or not os.path.exists(path):
-        return False
-    return get_cluster_active(filename=path) is not None
-
-
-def _parse_url(broker_url):
-    params = pika.URLParameters(broker_url)
-    return params.host
-
-
-def config_from_broker_urls(active, nodes):
-    settings = {
-        'active': _parse_url(active),
-        'nodes': [_parse_url(node) for node in nodes if node]
-    }
-    set_cluster_settings(settings)
-
-
 class ClusterHTTPClient(HTTPClient):
     default_timeout_sec = 5
     retries = 30
     retry_interval = 3
 
     def __init__(self, *args, **kwargs):
         super(ClusterHTTPClient, self).__init__(*args, **kwargs)
+        hosts = self.host if isinstance(self.host, list) else [self.host]
+        self.hosts = itertools.cycle(hosts)
```

Comment:
Probably good to randomise the order here so that we don't end up with one manager taking every single request until it goes down (a rudimentary sort of load balancing).